providers/supermicro/supermicro.go

```diff
@@ -136,14 +136,18 @@ func NewClient(host, user, pass string, log logr.Logger, opts ...Option) *Client
 		opt(defaultConfig)
 	}
 
-	serviceClient := newBmcServiceClient(
+	serviceClient, err := newBmcServiceClient(
 		host,
 		defaultConfig.Port,
 		user,
 		pass,
 		httpclient.Build(defaultConfig.httpClientSetupFuncs...),
 	)
 
+	if err != nil {
+		return nil
```

Comment:
Is this intentionally returning nil here?

Reply:
Yeah, ultimately in this case I was unsure what to do with the error...   I suppose an argument could be made that this is a marginal improvement to error handling in this case compared to previous, but that's likely a bit generous.

Reply:
ack, I'd suggest leaving a comment to indicate its intentional